### PR TITLE
fix Issue 15191 - DIP25: Taking address of ref return is not type che…

### PR DESCRIPTION
**Pull request by WalterBright**
…cked soundly

### CODE REVIEW

File: src/expression.d [hunk 1]
```diff
@@ -10662,6 +10662,25 @@ extern (C++) final class AddrExp : UnaExp
...
+        {
+            CallExp ce = cast(CallExp)e1;
+            if (ce.e1.type.ty == Tfunction)
+            {
+                TypeFunction tf = cast(TypeFunction)ce.e1.type;
+                if (tf.isref)
+                {
+                    if (sc.func && !sc.intypeof)
+                    {
+                        if (sc.func.setUnsafe())
```
Comment (andralex) [inline src/expression.d]:
```d
if (tf.isref && sc.func && !sc.intypeof && sc.func.setUnsafe())
{
    error(.........);
}
```

**Comment by WalterBright** [comment]
Did the work.

**Comment by andralex** [comment]
noice

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**